src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt

```diff
@@ -88,6 +88,59 @@ class PetDaoTest @Autowired constructor(private val sql: DSLContext) {
         assertEquals(newOwnerId, petDao.getAllPetsByCompanyId(companyId)[0].ownerId)
     }
 
+    //Jooq batch
+    @Test
+    fun `create multiple pets successfully`() {
+        petDao.createMultiplePets(listOf(petCreationRequest1, petCreationRequest2))
+        val expectedResult = petDao.getAllPetsByCompanyId(companyId)
+        assertEquals(expectedResult, petDao.getAllPetsByCompanyId(companyId))
+    }
+
+    @Test
+    fun `create multiple pets with empty list`() {
+        petDao.createMultiplePets(emptyList())
+        val expectedResult = petDao.getAllPetsByCompanyId(companyId)
+        assertTrue(expectedResult.isEmpty())
+    }
+
+    @Test
+    fun `create duplicate pets multiple times`() {
+        val petCreationRequests = listOf(petCreationRequest1, petCreationRequest1)
+        petDao.createMultiplePets(petCreationRequests)
+        val expectedResult = petDao.getAllPetsByCompanyId(companyId)
+        assertEquals(expectedResult, petDao.getAllPetsByCompanyId(companyId))
+    }
+
+    @Test
+    fun `adopt multiple pets successfully`() {
+        val ownerId = 3L
+        val id1 = petDao.createNewPet(petCreationRequest1)
+        val id2 = petDao.createNewPet(petCreationRequest2)
+        val pet1 = petCreationRequest1.extractToPet(id1)
+        val pet2 = petCreationRequest2.extractToPet(id2)
+        val updatedPet1 = pet1.copy(ownerId = ownerId)
+        val updatedPet2 = pet2.copy(ownerId = ownerId)
+        val pets = listOf(updatedPet1, updatedPet2)
+        petDao.adoptMultiplePets(pets, ownerId)
+        assertEquals(ownerId, pets[0].ownerId)
+        assertEquals(ownerId, pets[1].ownerId)
+    }
+
+    @Test
+    fun `adopt empty pets list`() {
+        val ownerId = 3L
+        val pets = emptyList<Pet>()
+        assertDoesNotThrow{ petDao.adoptMultiplePets(pets, ownerId) }
+    }
+
+    @Test
+    fun `adopt pets already adopted by the same owner`() {
+        val ownerId = 2L
+        val pet1 = Pet(1, "Tom", PetType.CAT, companyId, LocalDate.now(), ownerId)
+        val pet2 = Pet(2, "Luke", PetType.DOG, companyId, LocalDate.now(), ownerId)
+        assertDoesNotThrow{ petDao.adoptMultiplePets(listOf(pet1, pet2), ownerId) }
```

Comment:
Why do you expect an error not to be thrown? Does petDao.adoptMultiplePets ever throw exception?

Reply:
It an unnecessery test, I'll remove it